wgpu-core/src/validation.rs

```diff
@@ -1237,3 +1237,67 @@ impl Interface {
         Ok(outputs)
     }
 }
+
+fn address_space_matches(shader: naga::AddressSpace, binding: naga::AddressSpace) -> bool {
+    match (shader, binding) {
+        (
+            naga::AddressSpace::Storage {
+                access: access_shader,
+            },
+            naga::AddressSpace::Storage {
+                access: access_pipeline,
+            },
+        ) => access_shader & access_pipeline == access_shader,
```

Comment:
Could you add a comment about why we need this, and what it does (as it may not be immediately obvious to an observer.

```suggestion
        ) => (access_shader & access_pipeline) == access_shader,
```

I always confuse the order of operations here, so let's be extra clear

Reply:
> Could you add a comment about why we need this, and what it does (as it may not be immediately obvious to an observer.

I added `// Allow read- and write-only usages to match read-write layouts:` as a comment - looks ok or do you want to clarify/rephrase?

> I always confuse the order of operations here, so let's be extra clear

Absolutely, done now.